src/set_theory/ordinal/natural_ops.lean

```diff
@@ -0,0 +1,325 @@
+greater than `a' ♯ b` and `a ♯ b'` for `a' < a` and `b' < b`. The natural multiplication `a ⨳ b` is
+likewise recursively defined as the least ordinal such that `a ⨳ b ♯ a' ⨳ b'` is greater than
+`a' ⨳ b ♯ a ⨳ b'` for any `a' < a` and `b' < b`.
+
+These operations form a rich algebraic structure: they're commutative, associative, preserve order,
+have the usual `0` and `1` from ordinals, and distribute over one another.
+
+Moreover, these operations are the addition and multiplication of ordinals when viewed as
+combinatorial `game`s. This makes them particularly useful for game theory.
+
+Finally, both operations admit simple, intuitive descriptions in terms of the Cantor normal form.
+The natural addition of two ordinals corresponds to adding their Cantor normal forms as if they were
+polynomials in `ω`. Likewise, their natural multiplication corresponds to multiplying the Cantor
+normal forms as polynomials.
+
+# Implementation notes
+
+Given the rich algebraic structure of these two operations, we choose to create a type synonym
+`nat_ordinal`, where we provide the appropriate instances. However, to avoid casting back and forth
+between both types, we attempt to prove and state most results on `ordinal`.
+
+# Todo
+
+- Define natural multiplication and provide a basic API.
+- Prove the characterizations of natural addition and multiplication in terms of the Cantor normal
+  form.
+-/
+
+universes u v
+
+open function
+
+noncomputable theory
+
+/-- A type synonym for ordinals with natural addition and multiplication. -/
+def nat_ordinal : Type* := ordinal
+
+/-- The identity function between `ordinal` and `nat_ordinal`. -/
+@[pattern] def ordinal.to_nat_ordinal : ordinal ≃ nat_ordinal := equiv.refl _
+
+/-- The identity function between `nat_ordinal` and `ordinal`. -/
+@[pattern] def nat_ordinal.to_ordinal : nat_ordinal ≃ ordinal := equiv.refl _
+
+open ordinal
+
+namespace nat_ordinal
+
+variables {a b c : nat_ordinal.{u}}
+
+@[simp] theorem to_ordinal_symm_eq : nat_ordinal.to_ordinal.symm = ordinal.to_nat_ordinal := rfl
+@[simp] theorem to_ordinal_to_nat_ordinal (a : nat_ordinal) : a.to_ordinal.to_nat_ordinal = a := rfl
+
+instance : has_zero nat_ordinal := ⟨to_nat_ordinal 0⟩
+instance : inhabited nat_ordinal := ⟨0⟩
+instance : has_one nat_ordinal := ⟨to_nat_ordinal 1⟩
+instance : has_well_founded nat_ordinal := ordinal.has_well_founded
+instance : linear_order nat_ordinal := ordinal.linear_order
+
+@[simp] theorem to_ordinal_zero : to_ordinal 0 = 0 := rfl
+@[simp] theorem to_ordinal_one : to_ordinal 1 = 1 := rfl
+
+@[simp] theorem to_ordinal_eq_zero (a) : to_ordinal a = 0 ↔ a = 0 := iff.rfl
+@[simp] theorem to_ordinal_eq_one (a) : to_ordinal a = 1 ↔ a = 1 := iff.rfl
+
+@[simp] theorem to_ordinal_lt_iff : a.to_ordinal < b.to_ordinal ↔ a < b := iff.rfl
+@[simp] theorem to_ordinal_le_iff : a.to_ordinal ≤ b.to_ordinal ↔ a ≤ b := iff.rfl
+@[simp] theorem to_ordinal_inj : a.to_ordinal = b.to_ordinal ↔ a = b := iff.rfl
```

Comment:
Given all these, is it useful to have the order equivalence too?

Reply:
As in, upgrade `to_ordinal` and `to_ordinal_nat` from equivalences to order isomorphisms? That seems like a good idea.

Reply:
Right exactly